OpenSim/Actuators/VandenBogert2011Muscle.cpp

```diff
@@ -0,0 +1,1234 @@
+void VandenBogert2011Muscle::extendAddToSystem(SimTK::MultibodySystem& system)
+const
+{
+
+    Super::extendAddToSystem(system);
+
+    SimTK_ASSERT(isObjectUpToDateWithProperties(),
+                 "VandenBogert2011Muscle: Muscle properties not up-to-date");
+
+    addStateVariable("fiber_length");
+    addStateVariable("activation");
+
+}
+
+void VandenBogert2011Muscle::extendInitStateFromProperties(SimTK::State& s)
+const
+{
+    Super::extendInitStateFromProperties(s);
+
+        setActivation(s, getDefaultActivation());
+        setFiberLength(s, getDefaultFiberLength());
+}
+
+void VandenBogert2011Muscle::extendSetPropertiesFromState(const SimTK::State& s)
+{
+    Super::extendSetPropertiesFromState(s);
+
+        setDefaultActivation(getStateVariableValue(s,"activation"));
+        setDefaultFiberLength(getStateVariableValue(s,"fiber_length"));
+
+}
+
+
+void VandenBogert2011Muscle::
+computeStateVariableDerivatives(const SimTK::State& s) const
+{
+
+    double activ=getActivation(s);
+    // For now we will start with static guess
+    // TODO: figure out how to use previous state as guess
+    SimTK::Vector ydotInitialGuess(2);
+    ydotInitialGuess[0]=0.0;
+    ydotInitialGuess[1]=0.0;
+    SimTK::Vec2 sdot=calcSolveMuscle(s, activ, ydotInitialGuess);
+
+    setStateVariableDerivativeValue(s, "fiber_length", sdot[0]);
+    setStateVariableDerivativeValue(s, "activation", sdot[1]);
+
+
+}
+
+
+//=============================================================================
+// COMPUTATION
+//=============================================================================
+double VandenBogert2011Muscle::fiberLengthToProjectedLength
+        (double fiberLength , bool normalizedValues=0) const {
+
+    double pennAtOptFiberLength = getPennAtOptFiberLength();
+    double projFibLen = 0;
+
+    if (pennAtOptFiberLength < 0.01) {
+        projFibLen = fiberLength;
+    }
+
+    else {
+
+        double optimalFiberLength=getOptimalFiberLength();
+
+        if (normalizedValues) {fiberLength=fiberLength*optimalFiberLength;};
+
+
+        double muscleWidth = sin(
+                pennAtOptFiberLength);  //b is the fixed distance of the fiber
+        //perpindicular to the tendon (width)
+
+        /*TODO: muscleWidth as cache variable
+         * recalculated. Should be moved info muslceLengthInfo cache */
+
+        // The fiber can not be shorter than b; if it is, the projected length
+        // equation below will return a complex value.  It also physically can
+        // not happen (the fiber being shorter than the fixed width)
+        if (fiberLength >= muscleWidth) {
+            projFibLen = sqrt(pow(fiberLength, 2) - pow(muscleWidth, 2));
+        }
+        else {
+            projFibLen = SimTK::NaN;
+        }  //TODO: Doing this for now, but need to
+        // come up with a clamping scheme (Millard has one)
+
+        if (normalizedValues) {projFibLen=projFibLen/optimalFiberLength;};
+
+
+    };
+    return projFibLen;
```

Comment:
You can get rid of nested scopes by just returning early:

```cpp
if (pennAtOptFiberLen < 0.01) return fiberLength;
```

Then the rest of the method does not need to be indented and you don't need the `else {}`.